confidant/routes/credentials.py

```diff
@@ -201,8 +196,21 @@ def get_credential(id):
         include_credential_pairs = True
 
         if settings.ENABLE_SAVE_LAST_DECRYPTION_TIME:
-            credential.last_decrypted_date = datetime.now()
+            now = datetime.now()
+            credential.last_decrypted_date = now
             credential.save()
+            # Also try to save the archived credential to stay consistent
+            try:
+                archived_credential = Credential.get(
+                    '{}-{}'.format(id, credential.revision)
+                )
```

Comment:
Put both the saves in here, and add exception handling for the save failing in the list of exceptions.